app/controllers/email/VerifyEmailController.scala

```diff
@@ -73,6 +73,44 @@ class VerifyEmailController @Inject()(val emailVerificationService: EmailVerific
 
   }
 
+  def contactPrefShow: Action[AnyContent] = (blockAgentPredicate) { implicit user =>
+
+    if (appConfig.features.letterToConfirmedEmailEnabled()) {
+      extractSessionEmail(user) match {
+        case Some(email) => Ok(verifyEmailView(email, isContactPrefJourney = true))
+        case _ => Redirect(controllers.contactPreference.routes.ContactPreferenceRedirectController.redirect())
+      }
+    } else {
+        NotFound(errorHandler.notFoundTemplate)
+    }
+  }
+
+  def contactPrefSendVerification: Action[AnyContent] = blockAgentPredicate.async { implicit user =>
+
+    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(user.headers, Some(user.session))
+
+    if (appConfig.features.letterToConfirmedEmailEnabled()) {
+      extractSessionEmail(user) match {
+        case Some(email) =>
+          // TO DO - Add route to new update email and contact pref action
+          emailVerificationService.createEmailVerificationRequest(email, "").map {
+            case Some(true) => Redirect(routes.VerifyEmailController.contactPrefShow())
+            case Some(false) =>
+              logWarn(
+                "[VerifyEmailController][sendVerification] - " +
```

Comment:
Could you rename please?

```suggestion
                "[VerifyEmailController][contactPrefSendVerification] - " +
```

Reply:
Done. Good spot.